src/v/iceberg/schema_avro.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "iceberg/datatypes.h"
+
+#include <avro/CustomAttributes.hh>
+#include <avro/LogicalType.hh>
+#include <avro/Schema.hh>
+
+namespace iceberg {
+
+// Translates the given field/type to the corresponding Avro schema.
+// The resulting schema is annotated with Iceberg attributes (e.g. 'field-id',
+// 'element-id').
+avro::Schema field_to_avro(const nested_field& field);
+avro::Schema struct_type_to_avro(const struct_type&, const ss::sstring& name);
```

Comment:
would it be useful to have a converter class `concept` that we would implement for every schema that needs conversion ?

Reply:
Not a bad idea, but I'm honestly not sure. Offhand I'm not exactly sure what that concept would include.